lib/general/analysis.bash

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Check for boilerplate comments
+boilerplate() {
+    local comment='# *** PLEASE REMOVE THESE COMMENTS BEFORE SUBMITTING YOUR SOLUTION ***'
+    grep -Fxq "$comment" "$filename" \
+    && echo '{"comment": "bash.general.boilerplate_comment", "type": "actionable"}'
+}
+
+# Other check functions can go here:
+# ...
+
+
+# ------------------------------------------------------------
+# Merge results into analysis.json generated by the shellcheck analysis.
+
+comments_array_as_json() {
+    jq -n --jsonargs '{comments: $ARGS.positional}' "$@"
+}
+
+write_json() {
+    local -i shellcheck_comments
+    local temp
+
+    # How many shellcheck comments were found?
+    shellcheck_comments=$(jq '.comments | length' "$out_dir/analysis.json")
+
+    case "$shellcheck_comments,${#general_comments[@]}" in
```

Comment:
Wouldn't it be simpler to always merge the comments from both files? `jq` supports treating missing as null. I'm not sure how simple it would be to write the combination that handles empty.